light/provider/mock/mock.go

```diff
@@ -55,10 +56,16 @@ func (p *Mock) String() string {
 	return fmt.Sprintf("Mock{id: %s, headers: %s, vals: %v}", p.id, headers.String(), vals.String())
 }
 
-func (p *Mock) LightBlock(_ context.Context, height int64) (*types.LightBlock, error) {
+func (p *Mock) LightBlock(ctx context.Context, height int64) (*types.LightBlock, error) {
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case <-time.After(10 * time.Millisecond):
+	}
```

Comment:
What's the case for `time.After`? Can we add a godoc above the select statement please?

Reply:
This is just in the mock provider to better simulate an i/o call and give an opportunity for the context to be canceled. Maybe the `time.After` part isn't necessary for the tests to pass

Reply:
In tests, I may do this to trigger a context switch so that what ever goroutine may be running the `select` statement yields execution to the goroutine where the `cancel` is called. Not sure if it's necessary in this case or not.

Reply:
I tried rewriting the test to not create this waiting period by using `time.After` and instead just check `ctx.Err()` but in these cases the test doesn't pass

Reply:
I didn't realize this was a mock, no need for a godoc really IMO.